doc/add_datafit.rst

```diff
@@ -0,0 +1,120 @@
+:orphan:
+
+How to add a custom datafit
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. _how:
+
+Motivated by generalized linear models but not limited to it, ``skglm`` solves problems of the form
+
+.. math::
+      \hat{\beta} \in
+      \arg\min_{\beta \in \mathbb{R}^p}
+      F(X\beta) + \Omega(\beta)
+      := \sum_{i=1}^n f_i(X_i^{\top}\beta) + \sum_{j=1}^p \Omega_j(\beta_j)
```

Comment:
``X_i`` refer to i-th row of X, right?
to avoid confusion with our code, we better keep X_j to denote the columns of X instead.

suggestion: $X_i^{\top}\beta$  ------> $[X \beta]_i$